indexing-service/src/main/java/org/apache/druid/indexing/input/DruidInputSource.java

```diff
@@ -499,9 +500,9 @@ public static List<TimelineObjectHolder<String, DataSegment>> getTimelineForInte
     Collection<DataSegment> usedSegments;
     while (true) {
       try {
-        usedSegments = coordinatorClient.fetchUsedSegmentsInDataSourceForIntervals(
-            dataSource,
-            Collections.singletonList(interval)
+        usedSegments = FutureUtils.getUnchecked(
+            coordinatorClient.fetchUsedSegments(dataSource, Collections.singletonList(interval)),
+            true
```

Comment:
I wonder if we still need the retry logic in the catch block below. It would be handled by the new `CoordinatorClient` itself.

Reply:
Good point, I'll remove this part.